homeassistant/components/rejseplanen/config_flow.py

```diff
@@ -0,0 +1,147 @@
+"""Config flow for Rejseplanen integration."""
+
+from py_rejseplan.api.base import baseAPIClient as Rejseplanen
+import voluptuous as vol
+
+from homeassistant.config_entries import (
+    ConfigEntry,
+    ConfigFlow,
+    ConfigFlowResult,
+    ConfigSubentryFlow,
+    SubentryFlowResult,
+)
+from homeassistant.core import callback
+from homeassistant.helpers import config_validation as cv
+from homeassistant.helpers.selector import (
+    NumberSelector,
+    NumberSelectorConfig,
+    NumberSelectorMode,
+    TextSelector,
+    TextSelectorConfig,
+    TextSelectorType,
+)
+
+from .const import (
+    CONF_AUTHENTICATION,
+    CONF_DEPARTURE_TYPE,
+    CONF_DIRECTION,
+    CONF_NAME,
+    CONF_ROUTE,
+    CONF_STOP_ID,
+    DEFAULT_STOP_NAME,
+    DEPARTURE_TYPE_OPTIONS,
+    DEPARTURE_TYPE_TO_CLASS,
+    DOMAIN,
+)
+
+CONFIG_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_AUTHENTICATION): str,
+    }
+)
+
+CONFIG_STOP_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_STOP_ID): NumberSelector(
+            NumberSelectorConfig(
+                mode=NumberSelectorMode.BOX, min=1, max=999999999, step=1
+            ),
+        ),
+        vol.Optional(CONF_NAME, default=DEFAULT_STOP_NAME): str,
+        vol.Optional(CONF_DIRECTION, default=[]): TextSelector(
+            TextSelectorConfig(
+                type=TextSelectorType.TEXT,
+                multiple=True,
+            )
+        ),
```

Comment:
So as you explained before, this was generally used as the end of that line. I think we can be a bit smarter here.

1. If we need to define the start and end station of a route, I would not call this a stop, but rather a route. Having a start and end station sounds more logical than a stop id and a direction. (When I think about direction I think about left or right or compass directions.
2. If we can't define a destination that is not on the route, is it possible to fetch a list of all the stops along that route? This way I can select stop A, and it will list all the stops that I can reach with that. I think this would also make it way more foolproof.

Reply:
Can you elaborate a bit on what data you put in?

Reply:
I mean at this point I am not even worried about not copying the old integration, if we can do better today, I'd say let's do better.

So if I understand correctly we define a single station, and tell where we'd like to go to, so we know what leaves the station and whatnot, but we don't know anything about arrival. Correct?